src/core/skills/characters/yijiang2013/qiaoshui.ts

```diff
@@ -0,0 +1,267 @@
+import { CardType } from 'core/cards/card';
+import { EventPacker, GameEventIdentifiers, ServerEventFinder } from 'core/event/event';
+import { Sanguosha } from 'core/game/engine';
+import { AllStage, CardUseStage, PlayerPhase } from 'core/game/stage_processor';
+import { Player } from 'core/player/player';
+import { PlayerCardsArea, PlayerId } from 'core/player/player_props';
+import { Room } from 'core/room/room';
+import { TargetGroupUtil } from 'core/shares/libs/utils/target_group';
+import { ExtralCardSkillProperty } from 'core/skills/cards/interface/extral_property';
+import { ActiveSkill, TriggerSkill } from 'core/skills/skill';
+import { OnDefineReleaseTiming } from 'core/skills/skill_hooks';
+import { CommonSkill, PersistentSkill, ShadowSkill } from 'core/skills/skill_wrappers';
+import { TranslationPack } from 'core/translations/translation_json_tool';
+
+@CommonSkill({ name: 'qiaoshui', description: 'qiaoshui_description' })
+export class QiaoShui extends ActiveSkill {
+  public static readonly WIN: string = 'qiaoshui_win';
+  public static readonly LOSE: string = 'qiaoshui_lose';
+
+  public canUse(): boolean {
+    return true;
+  }
+
+  public numberOfTargets(): number {
+    return 1;
+  }
+
+  public isAvailableTarget(owner: PlayerId, room: Room, target: PlayerId): boolean {
+    return room.canPindian(owner, target);
+  }
+
+  public cardFilter(): boolean {
+    return true;
+  }
+
+  public isAvailableCard(): boolean {
+    return false;
+  }
+
+  public async onUse(): Promise<boolean> {
+    return true;
+  }
+
+  public async onEffect(
+    room: Room,
+    skillEffectSkill: ServerEventFinder<GameEventIdentifiers.SkillEffectEvent>,
+  ): Promise<boolean> {
+    const { fromId, toIds } = skillEffectSkill;
+    const { pindianRecord } = await room.pindian(fromId, toIds!);
+    if (!pindianRecord.length) {
+      return false;
+    }
+
+    if (pindianRecord[0].winner === fromId) {
+      room.setFlag(fromId, QiaoShui.WIN, true, true);
+    } else {
+      await room.skip(fromId, PlayerPhase.PlayCardStage);
+      room.setFlag(fromId, QiaoShui.LOSE, true, true);
+    }
+
+    return true;
+  }
+}
+
+@ShadowSkill
+@PersistentSkill()
+@CommonSkill({ name: QiaoShui.GeneralName, description: QiaoShui.Description })
+export class QiaoShuiWin extends TriggerSkill implements OnDefineReleaseTiming {
+  public afterLosingSkill(room: Room): boolean {
```

Comment:
what's your purpose here? It means the skill will be executed in the next FinishStage regardlessly